imagebuildah/build.go

```diff
@@ -55,56 +55,55 @@ type Mount = specs.Mount
 
 type BuildOptions = define.BuildOptions
 
-// BuildDockerfiles parses a set of one or more Dockerfiles (which may be
-// URLs), creates one or more new Executors, and then runs
-// Prepare/Execute/Commit/Delete over the entire set of instructions.
-// If the Manifest option is set, returns the ID of the manifest list, else it
-// returns the ID of the built image, and if a name was assigned to it, a
-// canonical reference for that image.
-func BuildDockerfiles(ctx context.Context, store storage.Store, options define.BuildOptions, paths ...string) (id string, ref reference.Canonical, err error) {
-	if len(paths) == 0 {
-		return "", nil, errors.Errorf("error building: no dockerfiles specified")
+// BaeImages extracts the set of base images from the specified containerfiles
+func BaseImages(ctx context.Context, options define.BuildOptions, containerfiles []string) ([]string, error) {
+	if len(containerfiles) == 0 {
+		return nil, errors.Errorf("error extracting: no containerfiles specified")
 	}
-	if len(options.Platforms) > 1 && options.IIDFile != "" {
-		return "", nil, errors.Errorf("building multiple images, but iidfile %q can only be used to store one image ID", options.IIDFile)
+	logger := loggerFromOptions(options)
+
+	dockerfiles, err := readDockerfiles(logger, options, containerfiles)
+	if err != nil {
+		return nil, err
 	}
 
-	logger := logrus.New()
-	if options.Err != nil {
-		logger.SetOutput(options.Err)
-	} else {
-		logger.SetOutput(os.Stderr)
+	baseImages, err := baseImages(containerfiles, dockerfiles, options.From, options.Args)
```

Comment:
(Absolutely non-blocking: The `containerfiles, dockerfiles` argument pair reads to me as if there were a deep semantic distinction between “container” and “docker” in this context. The previous `paths` vs. `files` naming was a bit simpler in this respect, though I could find something to quibble about with that naming as well :) )